### PR TITLE
🧪 Add test for ValidEnvName

### PR DESCRIPTION
**Pull request by ks1686**
🎯 **What:** The `ValidEnvName` function in `internal/schema/validate.go` was missing test coverage.
📊 **Coverage:** Added test cases covering:
  - Valid cases (uppercase, lowercase, mixed case letters, single/multiple underscores, with digits)
  - Invalid cases (empty strings, starting with digits, containing spaces/hyphens/dots/punctuation/unicode characters)
✨ **Result:** The test coverage for the schema package validation logic is improved, ensuring the environment variable validation behaves correctly for all expected and edge case inputs.

---
*PR created automatically by Jules for task [17378943208999447833](https://jules.google.com/task/17378943208999447833) started by @ks1686*

### CODE REVIEW

File: internal/schema/validate_test.go [hunk 1]
```diff
@@ -517,3 +517,43 @@ func TestLocateFields_NestedManagers(t *testing.T) {
...
+		{"single letter lowercase", "z", true},
+		{"single underscore", "_", true},
+		{"uppercase word", "FOO", true},
+		{"lowercase word", "bar", true},
+		{"mixed case", "FooBar", true},
+		{"with digits", "FOO_123", true},
+		{"with trailing digit", "FOO1", true},
+		{"starts with underscore", "_FOO", true},
+		{"underscore and digits", "_123", true},
+
```
[the remark this answers is not in the document]
Comment (ks1686) [reply internal/schema/validate_test.go]: CI/Test failed, needs to be addressed.
